homeassistant/components/modbus/validators.py

```diff
@@ -189,6 +191,26 @@ def number_validator(value: Any) -> int | float:
         raise vol.Invalid(f"invalid number {value}") from err
 
 
+def fixedRegList_validator(value: Any) -> list:
+    """Check if the number of registers for target temp. inserted is correct."""
+    if isinstance(value, int):
+        value = [value] * 7
+        return list(value)
+
+    if len(list(value)) == len(HVACMode):
```

Comment:
is the list() really needed ? its just confusing.

Reply:
@janiversen another motning idea was to rewrite the routine to make it generalist. It could be useful also for the future development. I'll remove the list() cast too

Reply:
Why is it a bad idea not to do list(var) on a var that is already a list()....you have removed the int part earlier.

Reply:
Yes, I have already optimized that routine and the other check about overlapping addresses.. I'm trying to understand your earlier comment about "the init" before to commit the new code.. But, I'm not so good at puzzles.. :)

Reply:
It's not a puzzle ! it's quite obvious init contains voluptuous definitions ....you actually already added a validator, so you know it.

Reply:
Yes, but the puzzle is not the init itself, but how to change the execution order of the validation calls.. You know the problem: actually duplicate_entity_validator is called before the validation of the entity, this means that it is called before CLIMATE_SCHEMA, so before that `CONF_TARGET_TEMP` has been validated and transformed to a list...